auth_options.go

```diff
@@ -194,8 +209,35 @@ func (opts *AuthOptions) ToTokenV3CreateMap(scope map[string]interface{}) (map[s
 			req.Auth.Identity.Token = &tokenReq{
 				ID: opts.TokenID,
 			}
+
+		} else if opts.ApplicationCredentialID != "" {
+			// Configure the request for ApplicationCredentialID authentication.
+			// https://github.com/openstack/keystone-specs/blob/master/specs/keystone/queens/application-credentials.rst#using-an-application-credential-to-obtain-a-token
+			// There are three kinds of possible application_credential requests
+			// 1. application_credential id + secret.
+			// 2. application_credential name + user_id + secret
+			// 3. application_credential name + user_name + user_domain_id/user_domain_name
```

Comment:
How about rewording this to:

1. application_credential id + secret
2. application_credential name + secret + user_id
3. application_credential name + secret + username + domain_id / domain_name

Secondly, since an app credential secret is always required (I _think_ it's required, right? https://github.com/openstack/keystoneauth/blob/c2ae9e298e2acdc4e653bce18735294cf4df2454/keystoneauth1/loading/_plugins/identity/v3.py#L270-L272), right below this line, let's make sure one is set:

```go
if opts.ApplicationCredentialSecret == "" {
  return nil, ErrAppCredMissingSecret{}
}
```

You can add the new error to the bottom of this file: https://github.com/gophercloud/gophercloud/blob/master/errors.go#L453